src/js/hooks/useQueryParamsListener.js

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useMemo } from 'react';
+
+import { useLocation } from 'react-router-dom';
+
+// Hook calling callback passed as an argument on changes in specified params
+const useQueryParamsListener = ({ callback, params }) => {
+  const { search } = useLocation();
+
+  const query = useMemo(() => new URLSearchParams(search), [search]);
+
+  const paramsDependencies = useMemo(
+    () =>
+      params.map((param) => query.get(param)).join('|'),
+    [query, params],
+  );
+
+  useEffect(() => {
+    callback();
+  }, [paramsDependencies]);
+};
+
+export default useQueryParamsListener;
```

Comment:
what's the flow behind it? what is it used for? and what is the `.join('|')` responsible for?

Reply:
`join` is to concat all of the params to one simple object to detect changes in params.
I had to implement detecting the changes in params this way because those filters have a lot of asynchronous operations and this fact led to the issues with sending requests with wrong parameters, so now it's reacting to changes in params.
I joined it with "|" but it can be something else.